### PR TITLE
KOGITO-5564 Use Quarkus PgPool instead of creating a new one

### PR DESCRIPTION
**Pull request by cristianonicolai**
Assembly:
- https://github.com/kiegroup/kogito-examples/pull/815
- https://github.com/kiegroup/kogito-runtimes/pull/1472
- https://github.com/kiegroup/kie-docs/pull/3812

### CODE REVIEW

**Comment by cristianonicolai** [comment]
@tiagodolphine this change would mean we rely on Quarkus datasource instead of creating a new PgPool for Quarkus. This would break current setup as `kogito.persistence.postgresql.connection.uri` is not used anymore. Opening for us to discuss before doing more changes to docs and examples.

File: kogito-codegen-modules/kogito-codegen-processes/src/main/java/org/kie/kogito/codegen/process/persistence/PersistenceGenerator.java [hunk 1]
```diff
@@ -500,43 +500,47 @@ private CompilationUnit generateProtostreamBaseMarshaller(TemplatedGenerator gen
...
-            //inserting DI annotations
-            context().getDependencyInjectionAnnotator().withConfigInjection(uriConfigParam, KOGITO_POSTGRESQL_CONNECTION_URI);
-            context().getDependencyInjectionAnnotator().withProduces(clientProviderMethod, true);
-            context().getDependencyInjectionAnnotator().withNamed(clientProviderMethod, KOGITO);
-            context().getDependencyInjectionAnnotator().withApplicationComponent(pgClientProducerClazz);
-
-            Optional<GeneratedFile> generatedPgClientFile = generatePersistenceProviderClazz(pgClientProducerClazz,
-                    new CompilationUnit(KOGITO_PROCESS_INSTANCE_PACKAGE).addType(pgClientProducerClazz));
-            generatedPgClientFile.ifPresent(generatedFiles::add);
+            if (context() instanceof SpringBootKogitoBuildContext) {
```
Comment (tiagodolphine) [inline kogito-codegen-modules/kogito-codegen-processes/src/main/java/org/kie/kogito/codegen/process/persistence/PersistenceGenerator.java:508]: So, we just create the Producer to inject a PgPool for Spring, and for Quarkus we rely on the PgPool already provided by the quarkus-reactive-pg-client extension, right?

Comment (cristianonicolai) [reply kogito-codegen-modules/kogito-codegen-processes/src/main/java/org/kie/kogito/codegen/process/persistence/PersistenceGenerator.java:508]: yep

**Comment by MarianMacik** [comment]
Do we even need `kogito.persistence.postgresql.connection.uri` if it can be configured by using separate configuration options like host, username and password?

**Review by MarianMacik — COMMENTED** [review]
Looks good to me, just one comment about the uri parameter.

**Comment by cristianonicolai** [comment]
@MarianMacik `kogito.persistence.postgresql.connection.uri` remains in use only for Springboot, for Quarkus we're dropping it completely.

**Review by MarianMacik — APPROVED** [review]

File: kogito-codegen-modules/kogito-codegen-processes/src/test/java/org/kie/kogito/codegen/process/persistence/PostgrePersistenceGeneratorTest.java [hunk 2]
```diff
@@ -40,14 +41,48 @@
...
-    KogitoBuildContext context = QuarkusKogitoBuildContext.builder()
-            .withApplicationProperties(new File(TEST_RESOURCES))
-            .withPackageName(this.getClass().getPackage().getName())
-            .withAddonsConfig(AddonsConfig.builder().withPersistence(true).build())
-            .build();
 
     @Test
-    void testGeneratedFiles() {
+    void testQuarkusGeneratedFiles() {
+        KogitoBuildContext context = QuarkusKogitoBuildContext.builder()
```
Comment (danielezonca) [inline kogito-codegen-modules/kogito-codegen-processes/src/test/java/org/kie/kogito/codegen/process/persistence/PostgrePersistenceGeneratorTest.java:47]: Minor note, you can use [this mechanism](https://github.com/kiegroup/kogito-runtimes/blob/main/kogito-codegen-modules/kogito-codegen-decisions/src/test/java/org/kie/kogito/codegen/decision/DecisionCodegenTest.java#L51) to make the test parametric and test all the different context

Comment (cristianonicolai) [reply kogito-codegen-modules/kogito-codegen-processes/src/test/java/org/kie/kogito/codegen/process/persistence/PostgrePersistenceGeneratorTest.java]: @danielezonca thanks, that simplified the test

**Comment by cristianonicolai** [comment]
Jenkins rerun optaplanner tests

**Review by tiagodolphine — APPROVED** [review]

**Pull request merged**